packages/dd-trace/src/exporters/common/request.js

```diff
@@ -93,15 +95,29 @@ function request (data, options, callback) {
   options.agent = isSecure ? httpsAgent : httpAgent
 
   const onResponse = res => {
-    let responseData = ''
+    const chunks = []
 
     res.setTimeout(timeout)
 
-    res.on('data', chunk => { responseData += chunk })
+    res.on('data', chunk => {
+      const bufferChunk = Buffer.isBuffer(chunk) ? chunk : Buffer.from(chunk)
+      chunks.push(bufferChunk)
+    })
     res.on('end', () => {
+      const buffer = Buffer.concat(chunks)
```

Comment:
If the the chunks are strings, isn't this slower now?

Reply:
possibly. Wouldn't that be the kind of thing that would show up in our performance tests though? I'll split the logic anyway. 

`request.js` is growing too much in conditional logic though

Reply:
added conditional logic in https://github.com/DataDog/dd-trace-js/pull/3992/commits/900888df73a32a65bcb87974b7c0095626267b69

Reply:
It would never be strings. It was only ever a string because appending it to the `responseData` string coerced the buffer to a string. Unless you use `stream.setEncoding(...)`, the `data` events will only ever be buffers.